### ISSUE TITLE
Feature: option to use `desc: Monitor` instead of port name

### ISSUE THREAD

**Issue by herrwusel** [nwg-piotr/nwg-displays#23]
Thanks for nwg-displays. This might be a hyprland specific wish, but it would be nice to have the option to use the monitors' description instead of the port name.
The background is that I have multiple workplaces with different monitor setups. Using the description would make using different setups more robust.

**Comment by nwg-piotr** [comment]
Nothing in the Hyprland wiki indicates that we can use generic names, but I've just tried:

```text
monitor=Dell Inc. DELL E2214H 2RK1Y4333FVU,1920x1080@60.0,1366x0,1.0
monitor=AU Optronics 0x315C,1366x768@60.02,0x312,1.0
```

...and it seems to work well. If so, your request is doable.

**Comment by herrwusel** [comment]
It is written above this [section](http://wiki.hyprland.org/Configuring/Monitors/#disabling-a-monitor) (scroll up a bit). It follows the syntax `desc: Monitor description`.

**Comment by nwg-piotr** [comment]
Thanks.

I'll add what you requested when I'm done with what I'm working on at the moment, probably this week.

**Comment by nwg-piotr** [comment]
It seem we have a problem. Setting default workspace for a monitor seems not to work with descriptions.

```
workspace=HDMI-A-1,1
```
 works.

```
workspace=AOC 2475WR F17H4QA000449,1
```
does not work.

Also

```
workspace=1,monitor:HDMI-A-1,default:true
```
works

```
workspace=1,monitor:AOC 2475WR F17H4QA000449,default:true
```
does not work.

Unless I'm doing something wrong, we've reached a dead end.

**Comment by nwg-piotr** [comment]
Yes, it does not work yet.

https://github.com/hyprwm/Hyprland/pull/2670

**Comment by nwg-piotr** [comment]
The problem on the Hyprland side has just been resolved by @MightyPlaza (thanks again!).

@herrwusel - The feature you requested must wait on the `desc` branch till the nearest Hyprland release. Of course you can test it roght now, if you install the development version of Hyprland.

![2023-07-12-033412_hypr_screenshot](https://github.com/nwg-piotr/nwg-displays/assets/20579136/6af67f07-edee-4e5b-9945-8315fd0d8bf3)

**Comment by herrwusel** [comment]
Thanks for your quick implementation. I'll give it a try and report back.

**Comment by nwg-piotr** [comment]
I discovered a minor bug. Will fix later today.
